pkg/util/certificates/certificates.go

```diff
@@ -25,3 +39,71 @@ func ValidateCertificateExpiration(certData []byte, renewalThreshold time.Durati
 
 	return true, nil
 }
+
+func CreateSelfSignedCertificate(domain string) (certPem []byte, privateKeyPem []byte, err error) {
+	serialNumber, err := rand.Int(rand.Reader, serialNumberLimit)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to generate serial number for tls certificate: %w", err)
+	}
+
+	privateKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	cert := &x509.Certificate{
+		SerialNumber: serialNumber,
+		Subject: pkix.Name{
+			Country:            []string{"AT"},
+			Province:           []string{"UA"},
+			Locality:           []string{"Linz"},
+			Organization:       []string{"Dynatrace"},
+			OrganizationalUnit: []string{"Operator Self-Signed"},
+			CommonName:         domain,
+		},
+
+		NotBefore: time.Now(),
+		NotAfter:  time.Now().Add(7 * 24 * time.Hour),
+
+		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		BasicConstraintsValid: true,
+	}
+
+	certBytes, err := x509.CreateCertificate(rand.Reader, cert, cert, privateKey.Public(), privateKey)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	privateKeyBytes, err := x509.MarshalECPrivateKey(privateKey)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return certBytes, privateKeyBytes, nil
+}
+
+func CreateP12CertificateSecretData(cert []byte, privateKey []byte) (map[string][]byte, error) {
+	var data map[string][]byte
+
+	certPem := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: cert})
+	privateKeyPem := pem.EncodeToMemory(&pem.Block{Type: "PRIVATE KEY", Bytes: privateKey})
+
+	p12Blocks, err := pkcs12.ToPEM(cert, certPassword)
```

Comment:
AG needs binary format of p12 file. Not sure if "golang.org/x/crypto/pkcs12" can provide it

Reply:
Fixed with new library! :smile: